kolibri/core/auth/utils/migrate.py

```diff
@@ -0,0 +1,244 @@
+        # Seems that this can only be changed from the command line,
+        # so seems safe to assume it will always be set to the default value.
+        min(calculate_max_sqlite_variables() // len(Model._meta.fields), 500)
+        if vendor == "sqlite"
+        else 750
+    )
+
+    Model.objects.bulk_create(objects, batch_size=batch_size)
+
+
+blocklist = set(["id", "_morango_partition"])
+
+
+def merge_users(source_user, target_user):  # noqa C901
+    """
+    Utility to merge two users. It makes no assumptions about whether
+    the users are in the same facility and does raw copies of all
+    associated user data, rather than try to do anything clever.
+    """
+    if source_user.id == target_user.id:
+        raise ValueError("Cannot merge a user with themselves")
+
+    _merge_user_models(source_user, target_user)
+
+    id_map = {
+        FacilityUser: {source_user.id: target_user.id},
+        FacilityDataset: {
+            source_user.dataset_id: target_user.dataset_id,
+        },
+    }
+
+    def _merge_log_data(LogModel):
+        log_map = {}
+        id_map[LogModel] = log_map
+        new_logs = []
+        related_fields = [f for f in LogModel._meta.concrete_fields if f.is_relation]
+        source_logs = LogModel.objects.filter(user=source_user)
+        target_log_ids = set(
+            LogModel.objects.filter(user=target_user).values_list("id", flat=True)
+        )
+        for log in source_logs:
+            # Get all serialializable fields
+            data = log.serialize()
+            # Remove fields that we explicitly know we don't want to copy
+            for f in blocklist:
+                if f in data:
+                    del data[f]
+            # Iterate through each relation and map the old id to the new id for the foreign key
+            for relation in related_fields:
+                data[relation.attname] = id_map[relation.related_model][
+                    data[relation.attname]
+                ]
+            # If this is a randomly created source id, preserve it, so we can stop the same logs
+            # being copied in repeatedly. If it is not random, remove it, so we can recreate
+            # it on the target.
+            if log.calculate_source_id() is not None:
+                del data["_morango_source_id"]
+            new_log = LogModel()
+            for field, value in data.items():
+                try:
+                    field_obj = LogModel._meta.get_field(field)
+                    if hasattr(field_obj, "from_db_value"):
+                        value = field_obj.from_db_value(value, None, None, None)
+                except FieldDoesNotExist:
+                    pass
+                setattr(new_log, field, value)
+            if not new_log._morango_source_id:
+                new_log.id = new_log.calculate_uuid()
+            else:
+                # Have to do this, otherwise morango will overwrite the current source id on the model
```

Comment:
Until https://github.com/learningequality/morango/issues/141 is resolved, at least!

Reply:
https://github.com/learningequality/morango/pull/142